tests/test_check_parallel.py

```diff
@@ -0,0 +1,387 @@
+""" Puts the check_parallel system under test """
+# Copyright (c) 2020 Frank Harrison <doublethefish@gmail.com>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+# pylint: disable=protected-access,missing-function-docstring,no-self-use
+
+import collections
+import os
+
+import pytest
+
+import pylint.interfaces
+import pylint.lint.parallel
+from pylint.checkers.base_checker import BaseChecker
+from pylint.lint import PyLinter
+from pylint.lint.parallel import _worker_check_single_file as worker_check_single_file
+from pylint.lint.parallel import _worker_initialize as worker_initialize
+from pylint.lint.parallel import check_parallel
+from pylint.testutils import TestReporter as Reporter
+
+
+def _gen_file_data(idx=0):
+    """ Generates a file to use as a stream """
+    filepath = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), "input", "similar1")
+    )
+    file_data = (
+        "--test-file_data-name-%d--" % idx,
+        filepath,
+        "--test-file_data-modname-%d--" % idx,
+    )
+    return file_data
+
+
+def _gen_file_datas(count=1):
+    file_infos = [_gen_file_data(idx) for idx in range(count)]
```

Comment:
The variable assignment can be removed:
```suggestion
    return [_gen_file_data(idx) for idx in range(count)]
```